### PR TITLE
[updates] add new architecture support

### PR DESCRIPTION
**Pull request by Kudo**
# Why

expo-updates does not support new architecture on ios because the way we create bridge is for the old architecture mode: https://github.com/expo/expo/blob/38ded1f9cf7fb69773bcd8126ede9441199e6594/packages/expo-updates/ios/EXUpdates/ReactDelegateHandler/ExpoUpdatesReactDelegateHandler.swift#L85-L88
close ENG-8455

# How

- introduce a `EXCreateReactBindingRootView` that could support new architecture. later we could maybe replace 
[DevClientAppDelegate](https://github.com/expo/expo/blob/main/packages/expo-dev-menu/ios/ReactNativeCompatibles/ReactNative/DevClientAppDelegate.mm) with this one.
- remove "bridge" from updates. originally we use bridge to send js events. with this pr, we could leverage expo-modules-core's event emitter to send js events. later we can benefit from bridgeless mode.

# Test Plan

- ci passed
- fabric-tester for the change to test on new architecture mode
- [UpdatesAPIDemo](https://github.com/expo/UpdatesAPIDemo) on new architecture mode

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]

**Review by gabrieldonadel — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

File: packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.kt [hunk 1]
```diff
@@ -160,58 +160,67 @@ object UpdatesUtils {
...
+    logger.info("Emitted event: name = $eventName, type = $eventType")
+    eventEmitter.emit(eventName, eventParams)
+  }
+
+  fun sendQueuedEventsToAppContext(
+    shouldEmitJsEvents: Boolean,
+    weakAppContext: WeakReference<AppContext>?,
+    logger: UpdatesLogger
+  ) {
+    if (!shouldEmitJsEvents) {
```
Comment (wschurman) [inline packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.kt:204]: should we have the same `logger.error` calls in these guard clauses as we do in the `sendEventToAppContext` guard clauses?

**Review by wschurman — APPROVED** [review]
I only reviewed the expo-updates/expo-updates-interface code thoroughly.  Other stuff looks fine too, but I'm less familiar with it.

**Comment by wschurman** [comment]
Thanks for doing this btw!

**Review by douglowder — APPROVED** [review]

File: packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.kt (hunk 1 above)
Comment (Kudo) [reply packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.kt:204]: for sending "queued" events, it looks like we didn't log when things are not ready: https://github.com/expo/expo/blob/23929f09c1b9c7f74390d4eba394a4a7a95585eb/packages/expo-updates/ios/EXUpdates/EnabledAppController.swift#L225-L233. i could add log for both ios and android though. wdyt?

Comment (Kudo) [reply packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesUtils.kt:204]: makes sense to me. thanks for sharing that!

**Pull request merged**